Modules/Synchronization/src/Synchronization.ConsumerApi/Controllers/SyncRunsController.cs

```diff
@@ -26,20 +29,28 @@ namespace Backbone.Modules.Synchronization.ConsumerApi.Controllers;
 public class SyncRunsController : ApiControllerBase
 {
     private readonly ApplicationOptions _options;
+    private readonly IUserContext _userContext;
+    private readonly IIdentitiesRepository _identitiesRepository;
 
-    public SyncRunsController(IMediator mediator, IOptions<ApplicationOptions> options) : base(mediator)
+    public SyncRunsController(IMediator mediator, IOptions<ApplicationOptions> options, IUserContext userContext, IIdentitiesRepository identitiesRepository) : base(mediator)
     {
         _options = options.Value;
+        _userContext = userContext;
+        _identitiesRepository = identitiesRepository;
     }
 
-
     [HttpPost]
     [ProducesResponseType(typeof(HttpResponseEnvelopeResult<StartSyncRunResponse>), StatusCodes.Status200OK)]
     [ProducesError(StatusCodes.Status400BadRequest)]
     public async Task<IActionResult> StartSyncRun(StartSyncRunRequestBody requestBody,
         [FromHeader(Name = "X-Supported-Datawallet-Version")]
         ushort supportedDatawalletVersion, CancellationToken cancellationToken)
     {
+        var identity = await _identitiesRepository.FindByAddress(_userContext.GetAddress(), CancellationToken.None);
+
+        if (identity!.Status == IdentityStatus.ToBeDeleted)
+            throw new ApplicationException(ApplicationErrors.SyncRuns.CannotStartSyncRunWhileIdentityIsToBeDeleted());
```

Comment:
As said before, this should be extracted in a method and reused like that.

Reply:
Changes made.